client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -41,51 +41,29 @@ const siteMigration: Flow = {
 			? [ STEPS.PICK_SITE, STEPS.SITE_CREATION_STEP, STEPS.PROCESSING ]
 			: [];
 
-		return [ ...baseSteps, ...hostedVariantSteps ];
+		return stepsWithRequiredLogin( [ ...baseSteps, ...hostedVariantSteps ] );
 	},
 
 	useAssertConditions(): AssertConditionResult {
 		const { siteSlug, siteId } = useSiteData();
-		const userIsLoggedIn = useSelect(
-			( select ) => ( select( USER_STORE ) as UserSelect ).isCurrentUserLoggedIn(),
-			[]
-		);
-		const startUrl = useStartUrl( this.variantSlug ?? FLOW_NAME );
-
-		let result: AssertConditionResult = { state: AssertConditionState.SUCCESS };
 		const { isOwner } = useIsSiteOwner();
 
-		useEffect( () => {
-			if ( ! userIsLoggedIn ) {
-				const logInUrl = startUrl;
-				window.location.assign( logInUrl );
-			}
-		}, [ startUrl, userIsLoggedIn ] );
-
 		useEffect( () => {
 			if ( isOwner === false ) {
 				window.location.assign( '/start' );
 			}
 		}, [ isOwner ] );
```

Comment:
I am seeing a race condition around this check for the flow. It feels more likely when I test using a URL like `/setup/site-migration/fr`.

I think we likely still need a check for the user's login state. It used to be implicit due to redirects before, but we should keep that in place.
```suggestion
		const userIsLoggedIn = useSelect(
			( select ) => ( select( USER_STORE ) as UserSelect ).isCurrentUserLoggedIn(),
			[]
		);

		useEffect( () => {
			if ( isOwner === false && userIsLoggedIn ) {
				window.location.assign( '/start' );
			}
		}, [ isOwner, userIsLoggedIn ] );
```

Reply:
Fixed here 7d080c58aea351810c3433d901f150e0649b4799